RecoJets/JetProducers/src/PileupJetIdAlgo.cc

```diff
@@ -207,15 +194,17 @@ void PileupJetIdAlgo::runMva()
                           if(std::abs(internalId_.jetEta_) > jEtaMax_.at(nEtaBins_-1)) {
                               internalId_.mva_ = -2.;
                           } else {
-                            for(int v=0; v<nEtaBins_;v++){
-                                if(std::abs(internalId_.jetEta_)>=jEtaMin_.at(v) && std::abs(internalId_.jetEta_)<jEtaMax_.at(v)){
-                                    internalId_.mva_ = etaReader_.at(v)->EvaluateMVA( tmvaMethod_.c_str() );
+                            for(int v=0; v<nEtaBins_; v++){
+                                if(std::abs(internalId_.jetEta_)>=jEtaMin_.at(v) && std::abs(internalId_.jetEta_)<jEtaMax_.at(v)) {
+                                    std::vector<float> vars = getMVAvars(tmvaEtaVariables_.at(v));
+                                    internalId_.mva_ = etaReader_.at(v)->GetClassifier(vars.data());
```

Comment:
With the change I suggested above to getMVAvars(), lines 199-200 become:
```
 internalId_.mva_ = getMVAvars(etaReader_.at(v), tmvaEtaVariables_.at(v));
```
You might also change the name of getMVAvars() to something like getMVAVal().